### PR TITLE
feature: ban participants – backend

### PR DESCRIPTION
**Pull request by BenedictHomuth**
## Description

This PR enhances the UpdateBoardSession with a field that indicates if a participant is banned. When that is the case, the participant is not able to join the board.

## Changelog

Created a new context, since we need to return an error while on the joinBoard route, so the frontend can catch it and do not allow for a banned participant to enter.

Created a new error for when a participant is banned (403 – Gone). I choose this one since 403 - Forbidden is already in use, when the participant enters the wrong passphrase.

Enhanced the updateBoardSession Route, with the new banned field.

Added test to verify that the updated banned field works.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] The light- and dark-theme are both supported and tested
- [ ] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

### CODE REVIEW

File: server/src/api/context.go [hunk 1]
```diff
@@ -53,7 +52,35 @@ func (s *Server) BoardParticipantContext(next http.Handler) http.Handler {
...
 		user := r.Context().Value("User").(uuid.UUID)
+		banned, err := s.sessions.ParticipantBanned(r.Context(), board, user)
+		if err != nil {
+			log.Errorw("unable to check if participant is banned", "err", err)
+			common.Throw(w, r, common.InternalServerError)
+			return
+		}
+
+		if banned {
+			common.Throw(w, r, common.GoneError(errors.New("participant is currently banned from this session")))
```
Comment (MarvinRoth) [inline server/src/api/context.go:63]: Im not sure if GoneError is the most suitable for a banning error. Maybe it should be something like access denied.
Gone would indicate that a ressource is gone for good and not just that the access to the ressource is not granted

File: server/src/common/error.go [hunk 2]
```diff
@@ -40,6 +41,15 @@ func ForbiddenError(err error) *APIError {
 	}
 }
 
+func GoneError(err error) *APIError {
```
Comment (MarvinRoth) [inline server/src/common/error.go:44]: just as mentioned in server/src/api/context.go, maybe not the right error for the job

**Review by MarvinRoth — COMMENTED** [review]
Maybe reconsider the error type. The business logic seems to work as intended

File: server/src/api/context.go (hunk 1 above)
Comment (brandstetterm) [reply server/src/api/context.go]: But the 403 Forbidden is better than 401 Unauthorized. 
401 Unauthorized is specifically for the case that the client is unauthenticated, which is not the case here.

**Review by brandstetterm — APPROVED** [review]
lgtm 👍🏼

File: server/src/api/context.go (hunk 1 above)
Comment (BenedictHomuth) [reply server/src/api/context.go]: 403 won in the end! :)

**Pull request merged**